### DISCUSSION TITLE
API references and keys

### DISCUSSION THREAD

**Discussion by OriAshkenazi** [AntonOsika/gpt-engineer#246] [Ideas]
Given my app implementation requires APIs that the model doesn't know how to use properly, and doesn't know the specific API keys needed for a clear run of the code - Where is the right place to put the API's documentation and keys for context?
I was thinking of putting it all in `identity/philosophy`, but am not sure.

**Comment by OriAshkenazi** [comment]
Made an experiment by pasting the required API documentation into `identity/philosophy`, it made GPT skip implementing almost all of the functions, in all classes!
I'd also mention I'm currently using 3.5-turbo and not 4, this might be a contributing factor to the skipping, caused by less context due to a lack of token attention span...

Nevertheless, I think it's a good idea to create a separate identity file to include API use guidelines and personal assets like API keys.
The guideline's format should also be considered, is the full documentation needed? or can we just provide code snippets? maybe both are required?

**Reply by clickbrain** [reply]
I have also run into this and that would be great.
